### ISSUE TITLE
Music (Songs) not added / parsed correctly into Albums

### ISSUE THREAD

**Issue by hasezoey** [jellyfin/jellyfin#6694]
**Describe the bug**
Songs not correctly parsed into correct Albums

**System (please complete the following information):**
 - OS: Ubuntu Server 21.04 (Jellyfin server), Manjaro 21.1.4 Firefox (Client)
 - Virtualization: Docker
 - Clients: Browser
 - Browser: Firefox 93
 - Jellyfin Version: Jellyfin Docker 10.8.0 (`jellyfin/jellyfin:unstable@sha256:a2a350f5bce7a5530f33fd65ff1774027d1692f0d804d01bf74a4cc51ba60aaf`)
 - FFmpeg Version: Whatever is in the docker by default
 - Playback: Not applicable to this issue
 - Hardware Acceleration: Not applicable to this issue
 - Installed Plugins: MusicBrainz
 - Reverse Proxy: nginx
 - Base URL: yes, Not applicable to this issue
 - Networking: Docker (Host)
 - Storage: Local (Docker bind mount)

**To Reproduce**
Have folder structure:
```txt
$ tree .
./
├── Album 1
│   ├── 1-01 Title1.mp3
│   ├── 1-02 Title2.mp3
│   ├── 1-03 Title3.mp3
│   ├── 1-04 Title4.mp3
│   ├── 1-05 Title5.mp3
│   ├── album.nfo
│   └── cover.jpg
... many more albums like this
└──  Unsorted Songs
    ├── SomeSong1.mp3
    ├── SomeSong2.mp3
    ├── SomeSong3.mp3
    ├── album.nfo
    └── cover.jpg
... and many more unsorted songs
```

**Expected behavior**
That each song (that has at the required tags, like `Album`) to be sorted in its own / correct Album (inside jellyfin's library, not touching the actual file on the filesystem)

**Server Logs**

<details>
<summary>Server logs</summary>

```txt
[16:23:36] [INF] [1] Main: Jellyfin version: 10.8.0,
[16:23:36] [INF] [1] Main: Environment Variables: ["[JELLYFIN_FFMPEG, /usr/lib/jellyfin-ffmpeg/ffmpeg]", "[JELLYFIN_CACHE_DIR, /cache]", "[JELLYFIN_DATA_DIR, /config]", "[JELLYFIN_MEDIA_DIR, /media]", "[JELLYFIN_CONFIG_DIR, /config/config]", "[JELLYFIN_WEB_DIR, /jellyfin/jellyfin-web]", "[JELLYFIN_LOG_DIR, /config/log]", "[DOTNET_SYSTEM_GLOBALIZATION_INVARIANT, 1]"],
[16:23:36] [INF] [1] Main: Arguments: ["/jellyfin/jellyfin.dll"],
[16:23:36] [INF] [1] Main: Operating system: Linux,
[16:23:36] [INF] [1] Main: Architecture: X64,
[16:23:36] [INF] [1] Main: 64-Bit Process: True,
[16:23:36] [INF] [1] Main: User Interactive: True,
[16:23:36] [INF] [1] Main: Processor count: 8,
[16:23:36] [INF] [1] Main: Program data path: /config,
[16:23:36] [INF] [1] Main: Web resources path: /jellyfin/jellyfin-web,
[16:23:36] [INF] [1] Main: Application directory: /jellyfin/,
[16:23:36] [INF] [1] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Setting cache path: /cache,
[16:23:37] [INF] [1] Emby.Server.Implementations.ApplicationHost: Loading assemblies,
[16:23:37] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Jellyfin.Plugin.AniList, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null from /config/plugins/AniList_4.0.0.0/Jellyfin.Plugin.AniList.dll,
[16:23:37] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Jellyfin.Plugin.CoverArtArchive, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Cover Art Archive_3.0.0.0/Jellyfin.Plugin.CoverArtArchive.dll,
[16:23:37] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly TvDbSharper, Version=3.2.2.0, Culture=neutral, PublicKeyToken=null from /config/plugins/TheTVDB_6.0.0.0/TvDbSharper.dll,
[16:23:37] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Jellyfin.Plugin.Tvdb, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null from /config/plugins/TheTVDB_6.0.0.0/Jellyfin.Plugin.Tvdb.dll,
[16:23:37] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Defined LAN addresses : [::1/128,fc00::/7,fe80::/10,127.0.0.1/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16],
[16:23:37] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Defined LAN exclusions : [],
[16:23:37] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using LAN addresses: [::1/128,fc00::/7,fe80::/10,127.0.0.1/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16],
[16:23:37] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using bind addresses: [],
[16:23:37] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using bind exclusions: [],
[16:23:37] [ERR] [1] Jellyfin.Networking.Manager.NetworkManager: Unable to parse bind override: https://jellyfin.sylvi,
[16:23:40] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: AniList 4.0.0.0,
[16:23:40] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: Cover Art Archive 3.0.0.0,
[16:23:40] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: TheTVDB 6.0.0.0,
[16:23:40] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: TMDb 10.8.0.0,
[16:23:40] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: OMDb 10.8.0.0,
[16:23:40] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: MusicBrainz 10.8.0.0,
[16:23:40] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: AudioDB 10.8.0.0,
[16:23:40] [INF] [1] Main: Kestrel listening on All Addresses,
[16:23:40] [WRN] [1] Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager: No XML encryptor configured. Key {6ee4a90d-e90f-49e5-a38d-62754e6d283d} may be persisted to storage in unencrypted form.,
[16:23:40] [INF] [1] Emby.Server.Implementations.ApplicationHost: Running startup tasks,
[16:23:40] [INF] [1] Emby.Server.Implementations.ScheduledTasks.TaskManager: Daily trigger for Extract Chapter Images set to fire at 2021-10-13 02:00:00.000 +00:00, which is 09:36:19.1330253 from now.,
[16:23:41] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Found ffmpeg version 4.3.2,
[16:23:41] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available decoders: ["h264", "h264_qsv", "h264_cuvid", "hevc", "hevc_qsv", "hevc_cuvid", "mpeg2video", "mpeg2_qsv", "mpeg2_cuvid", "mpeg4", "mpeg4_cuvid", "msmpeg4", "vc1_qsv", "vc1_cuvid", "vp8_cuvid", "vp8_qsv", "vp9_cuvid", "vp9_qsv", "aac", "ac3", "flac", "mp3"],
[16:23:41] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available encoders: ["libx264", "h264_amf", "h264_nvenc", "h264_qsv", "h264_v4l2m2m", "h264_vaapi", "libx265", "hevc_amf", "hevc_nvenc", "hevc_qsv", "hevc_vaapi", "mpeg4", "msmpeg4", "libvpx", "libvpx-vp9", "aac", "ac3", "flac", "libmp3lame", "libopus", "libvorbis", "srt"],
[16:23:41] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available filters: ["hwupload_cuda", "overlay_cuda", "scale_cuda", "tonemap_opencl", "tonemap_vaapi", "yadif_cuda"],
[16:23:41] [WRN] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Filter: tonemap_cuda with option GPU accelerated HDR to SDR tonemapping is not available,
[16:23:41] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Available hwaccel types: ["vdpau", "cuda", "vaapi", "qsv", "drm", "opencl"],
[16:23:41] [INF] [1] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: FFmpeg: /usr/lib/jellyfin-ffmpeg/ffmpeg,
[16:23:41] [INF] [1] Emby.Server.Implementations.ApplicationHost: ServerId: 0411b444f3724b91bc87f05bbbb2b883,
[16:23:41] [INF] [1] Emby.Dlna.Main.DlnaEntryPoint: Registering publisher for urn:schemas-upnp-org:device:MediaServer:1 on 192.168.2.6/24,
[16:23:41] [INF] [1] Emby.Dlna.Main.DlnaEntryPoint: Registering publisher for urn:schemas-upnp-org:device:MediaServer:1 on 172.17.0.1/16,
[16:23:41] [INF] [1] Emby.Dlna.Main.DlnaEntryPoint: Registering publisher for urn:schemas-upnp-org:device:MediaServer:1 on 172.20.0.1/16,
[16:23:41] [INF] [1] Emby.Server.Implementations.ApplicationHost: Executed all pre-startup entry points in 0:00:00.0994913,
[16:23:41] [INF] [1] Emby.Server.Implementations.ApplicationHost: Core startup complete,
[16:23:41] [INF] [1] Emby.Server.Implementations.IO.LibraryMonitor: Skipping realtime monitor for /media/Movies_test/Movies_test because the path does not exist,
[16:23:41] [INF] [18] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /media/Pictures,
[16:23:41] [INF] [1] Emby.Server.Implementations.EntryPoints.ExternalPortForwarding: Starting NAT discovery,
[16:23:41] [INF] [10] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /media/Shows,
[16:23:41] [INF] [21] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /media/Music,
[16:23:41] [INF] [1] Emby.Server.Implementations.ApplicationHost: Executed all post-startup entry points in 0:00:00.1845739,
[16:23:41] [INF] [1] Main: Startup complete 0:00:04.947204,
[16:23:43] [INF] [8] Emby.Server.Implementations.ScheduledTasks.TaskManager: StartupTrigger fired for task: Update Plugins,
[16:23:43] [INF] [8] Emby.Server.Implementations.ScheduledTasks.TaskManager: Queuing task PluginUpdateTask,
[16:23:43] [INF] [13] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing Update Plugins,
[16:23:45] [INF] [19] Emby.Server.Implementations.ScheduledTasks.TaskManager: Update Plugins Completed after 0 minute(s) and 1 seconds,
[16:23:45] [INF] [19] Emby.Server.Implementations.ScheduledTasks.TaskManager: ExecuteQueuedTasks,
[16:23:50] [INF] [33] Emby.Server.Implementations.HttpServer.WebSocketManager: WS 192.168.2.8 request,
[16:23:50] [INF] [33] Emby.Server.Implementations.HttpServer.WebSocketManager: WS 192.168.2.8 closed,
[16:23:51] [INF] [19] Emby.Server.Implementations.HttpServer.WebSocketManager: WS 192.168.2.8 request
```

</details>

**FFmpeg Logs**
Not applicable to this issue

**Browser Console Logs**
Not applicable to this issue

**Screenshots**

**Additional context**
some songs have the required metadata, some dont (like `Title` or `Album`)

the songs that have `Album` set (either with file or directly in jellyfin), do not get sorted over to their own album, but stay in the `Unsorted Songs` "Album"

i already tried:
- to delete `album.nfo` from the folder
- to run a re-scan of the library (modes: `Replace all metadata` & `Scan for new & updated files`) (both on the `Unsorted Songs` "Album" and on a library level)

Note: the metadata is added with Musicbrainz Picard and ffmpeg

also, i expected this to work because of [Folder Structure for Music Documentation](https://jellyfin.org/docs/general/server/media/music.html) stating that it should work:

> You can also separate your music into artist folders or with no folder structure at all.
> ...
> Individual songs have no required parameters for filenames since the information will be scraped from metadata.

**Comment by hasezoey** [comment]
also #5172 may be related, but it didnt match my case so i opened a new issue

**Comment by michael9dk** [comment]
I had issues when the AlbumArtist tag was present. I removed it with Picard, and kept only the single Album tag. 
For Compilations I added AlbumArtist as "Various artists", and removed Artists (notice the s).
For folders with a bunch of mixed files, eg Dance, I added AlbumArtist tag as "Dance music", to make it appear as a huge compilation.

**Comment by hasezoey** [comment]
this is still a issue from what i can tell in jellyfin version `jellyfin/jellyfin:unstable@sha256:e3f042c449a8dcfb0cde999d7bb2e164cf9f4532f277d4f97fbdd8089ec5d3c8`

**Comment by dschrempf** [comment]
I also have this issue with the following folder structure:
```
./Root
./Root/Album1/...
./Root.Album2/...
./Root/AlbumX/...
./Root/SingletonSong.flac
```

All songs in `/Root/` get sorted into the album of `/Root/SingletonSong.flac` although the albums are tagged correctly.

**Comment by dschrempf** [comment]
Apart from #4968, I also found #3169, which may be related.

**Comment by exaveal** [comment]
My problem is that all Songs without Album Tag directly in the folder of the artist are not listed on the artist info page. albums are listed. singles without an album not.

**Comment by dschrempf** [comment]
Has there been any progress on this? I wonder why not more people are hit by this issue. I do not think the mentioned folder structure is a very peculiar one.

**Comment by soup-eater** [comment]
> as a seperate issue: when having too many artists for a "ablum" (or in this case, folder), they obstruct other stuff, see:
> 
> Please ignore the names, some of them are screwed up because of the mentioned main issue
> 
> ![Screenshot_20211013_132345](https://user-images.githubusercontent.com/10911626/137123785-f06dae44-32ce-4aaf-a94a-b14a420a1179.png)

Im having the same issue except it freezes my page for a some seconds  (similar on android client)

**Comment by darkuranium** [comment]
Not much to update, it's the exact same problem / behaviour.

**Comment by darkuranium** [comment]
This is getting silly. The problem still exists, and it's been years. Any updates?

**Comment by felix920506** [comment]
Tagging as confirmed since this is actual behavior confirmed with devs.

**Comment by ItMeCorban** [comment]
I am so confused as to how this is still an issue, ya'll really can't just add a tracks section to an artists page? How was that missed in the first place?

**Comment by wacko37** [comment]
!!!BUMP on this!!! - I've only just realised this to be case, so bloody annoying. I listen to  a lot of DJ mixes and this means I have to go through every Artist folder and add individual mixes into there own folder. All my metadata is correct too, this make no sense at all.

**Comment by philipp2310** [comment]
Half way to becoming closed again as stale - giving another bump.
Issue still persists, Music without an album is currently not visible anywhere but in playlists. It IS playable by clicking play on the "empty" artists page. as ItMeCorban mentioned, a "track section" in the artists page, either with all songs, including album, or just the albumless songs is highly required.

Thank you for the awesome software! Wouldn't have added to the issue, but with the current digital "one song at a time releases", creating hundreds of 1 song almbums is a nightmare.

**Comment by medallyon** [comment]
Indeed, would love to be able to display "singles" not as part of albums (which also pollutes the "Albums" page) but as singles on Artist pages, and perhaps simply sorted under the "Songs" page.

**Comment by tibe-31** [comment]
It is much surprising that such an ultra basic feature is missing in a music browser : 
to access all the music files in a directory (however the folder structure is)
I have never seen this before on any other products dedicated to play music files, all the more when all meta tags are religiously set on all files.
Hope JF devs mind that they probably cut JF definitely from many potential users just because of this little thing.

Would it be so complicated, in the algo, when there is no album defined, to just consider there is a virtual one like 'Various" and link all "non album" songs to it:
./Root
./Root/Artist1/Album1/...
./Root/Artist1/Album2/...
./Root/Artist1/AlbumX/...
./Root/Artist1/***Various***/...

The same for artists : 
./Root
./Root/Artist1/...
./Root/Artist2/...
./Root/Artist3/...
./Root/***Various***/...

Many other solutions exist of course.

I am a bit shocked.

**Comment by ItMeCorban** [comment]
> Hope JF devs mind that they probably cut JF definitely from many potential users just because of this little thing.

This. I have tried jellyamp/other jellyfin music players and they work great, but they're unusable because half my library is missing. This is the main reason I still use plexamp and cannot recommend anyone use jellyfin if their primary use is music. A plexpass is worth the price for access to plexamp because my media is actually visible there.

**Comment by tibe-31** [comment]
If it can help someone, here is a non perfect (sad) workaround :
- create, under each artist folder, a subfolder called "Various" (don't use name "Other" as I did first : there is another bug here in JF, it seems to be a wtf keyword that makes the directory ignored if you do)
- for all "no album linked" music files of the artist, move them to this subfolder
- for all "no album linked" music files of all artists, tag the "Album" field with value "Various"

It's a bit work but then you can browse & read all your music files from the JF client (go to the "Various" album item under each artist item)

But it is sad definitely.

**Comment by dschrempf** [comment]
Actually, I did the same. You can configure beets to do this automatically for you. Something like

```
        singleton = "$artist/singletons/$title%sunique{}";
```

**Comment by Delivator** [comment]
Trying to migrate my playlists from Spotify to Jellyfin and this is breaking my neck at the moment.

**Comment by darkuranium** [comment]
I've honestly given up on Jellyfin for music, mostly because of this issue. And kind of just about given up on using it for video too, since syncplay doesn't seem to be working correctly, at least for me (and this was more or less the main reason as to why I wanted to try Jellyfin).

**Comment by doobes** [comment]
Geeze,

This is surprising.  I was wondering why my album-less stuff wasn't appearing.

The "Various" thing seems to work, but what a 9 line cluge.

Any comment from the dev's on this cluster?

**Comment by dschrempf** [comment]
I begin to suspect that this thread is just going unnoticed. Should we become active and inform somebody specific?

**Comment by felix920506** [comment]
This is a low priority item for Jellyfin right now, that's why it is not getting much attention.

We already tagged this issue confirmed. Additional comments won't make things go faster.
If this really bothers you that much then I suggest you go ahead and send a PR to fix this issue.

**Comment by Delivator** [comment]
"low priority" and "comments won't make things go faster". So if a lot of people have an issue, you wouldn't consider changing the priority?
Of course it would be nice if someone here could just fix it, but then it would probably... be fixed.

**Comment by doobes** [comment]
https://www.forbes.com/sites/forbescoachescouncil/2018/05/16/communicating-respect-know-your-audience/

**Comment by maru801** [comment]
> So if a lot of people have an issue, you wouldn't consider changing the priority?

There doesn't exist a definite list of issues sorted by priority. What was meant by "low-priority" was that out of those working on PRs for Jellyfin, their focus is currently on other things due to their personal interests.

If for example you or anyone else took up the mantle and attempted to create this functionality, it would be your or their priority to fix.

Also if it wasn't said before and to add to this issue thread, this behavior originates from Emby since Jellyfin forked from it before. You need to sort your music under album folders as described in the Jellyfin music docs.

**Comment by gnattu** [comment]
> Also if it wasn't said before and to add to this issue thread, this behavior originates from Emby since Jellyfin forked from it before. You need to sort your music under album folders as described in the Jellyfin music docs.

I hate this way as much as any other people in this issue complaining because the Emby way makes no sense to me at all. Why each music track has to have an album and why having an enforcement on folder structure when the music files has almost all metadata tagged in the file? I wanted to change that but it was not that easy. This means we need to complete rewrite the audio scanning logic to “fix” this issue and is ultimately blocked by the database refactor we are having now: #12798.

The old Emby design in the database enforces everything must have a physical folder and music tracks has to have an album as their parents. A lot has to be changed from the database schema perspective to support folder structure and album free music tracks as well. 

To sum it up, there are a lot to be worked on and essentially means to discard all Emby era code to write the Music library from scratch. I know current design is bad, but unfortunately we cannot work on this as high priority due to the amount of the work required and actual blockers we are having.

**Comment by dschrempf** [comment]
If the database structure enforces a connection from songs to albums, I am pretty sure this has a reason (maybe in the frontend for display purposes?). One could just collect all singleton tracks in a folder to a fake album unique to this folder.

**Comment by maru801** [comment]
> So whoever initially designed this functionality just ignore that not every song belongs to an album? Talk about oversights...

Correct me if I'm wrong, but albums don't always need to actual albums. Like for my music album, I add singles into their own album folder.

I think that's what Emby was going for with the structure of having it be sorted as folders of artists each with subfolders of their albums.

The issue here is that Emby used the folders to determine what album each song belongs to.

As previously stated, there's already desire to change this to use the embedded tags, but that will require large changes to the fundamental code that currently makes Jellyfin. The main devs already have their free time dedicated to rewriting the Jellyfin database structure which in itself is another huge workload.


> By the sounds of it, Jellyfin needs a roadmap or some kind of central prioritisation for who's working on what because having devs prioritise things based on what they feel like working on doesn't seem particularly scalable or efficient.

There already are roadmaps on GitHub. For example, [this one for v10.0](https://github.com/orgs/jellyfin/projects/55).

[You can find other similar roadmaps here](https://github.com/orgs/jellyfin/projects?query=is%3Aopen).

It's just that there isn't a main "todo" board that gets selectively sorted by some arbitrary priority ranking to tell all potential devs to work on something.

Also, people that work on Jellyfin have their own area of interests and knowledge. It's kind of a hard thing to tell all devs to focus on a thing when some might not be familiar with that side of things. And Jellyfin isn't a big organization. People come and go all the time providing PRs in their free time while having to also work another job.

**Comment by hasezoey** [comment]
As a update from my side (the OP of this issue):
- The UI [overflow issue](https://github.com/jellyfin/jellyfin/issues/6694#issuecomment-942202243) has been fixed. (by limiting the display and adding `and XX other artists.`)
- The [documentation issue](https://github.com/jellyfin/jellyfin/issues/6694#issue-1024031904)(at the bottom) which noted that unsorted songs are supported has been removed, from what i can tell.
- Unsorted songs in a single folder is still a issue (just tried putting some files with title, artist and album tags set into a single mixed folder).
- Album folders can be placed anywhere and they will get recognized correctly. (as long as they only contain that album's files)

TL;DR: the main issue persists, but it is not listed as "supported" in the docs anymore.

---

([re this comment](https://github.com/jellyfin/jellyfin/issues/6694#issuecomment-2518666986))

As a note for anyone that might be doing the same, i am using Musicbrainz-Picard (`Preset 3: [album artist]/[album]/[disc and track #] [artist] - [title]`), and this works greatly with jellyfin's folder structure.

For context, my folder structure now basically is:
```sh
$ tree .                   
.
├── ArtistA
│   ├── AlbumA
│   │   ├── 01 somefile.mp3
│   │   └── album.nfo
│   ├── AlbumC
│   │   ├── 01 somefile.mp3
│   │   └── album.nfo
│   ├── album.nfo
│   ├── artist.nfo
│   └── standaloneA.mp3
├── ArtistB
│   ├── AlbumB
│   │   ├── 01 somefile.mp3
│   │   └── album.nfo
│   ├── album.nfo
│   ├── artist.nfo
│   └── standaloneB.mp3
└── Unsorted
    ├── album.nfo
    └── something without tags.mp3
```

---

Jellyfin `10.10.3`(docker)
Picard `2.12.3`

**Comment by WindFreaker** [comment]
Hi, is there any recent updates to this issue? Jellyfin's persistence with sorting singles into each individual folder would require a major change to my library.

**Comment by dschrempf** [comment]
I don't know why nobody is picking this up either. I think it is an important feature. There must be some internal hurdles that we can not see.

**Comment by hyperdotbat** [comment]
Slightly unrelated, but lets say we do sort singles into folders, I personally dont like there is no separation between Album and Single simply based on it being a single song, an option to allow this would be great
EDIT: I have found the [feature request on Fider](https://features.jellyfin.org/posts/2220/group-music-releases-by-type-album-single-live-compilation-etc)

**Comment by dschrempf** [comment]
(I don't like it when a comment with a genuine question gets hidden as off topic).

I rephrase the question:
Are there hurdles/problems/issues that keep us from implementing this request? Is there a particular reason why we have not implemented this request yet? The issue is 5 years old.

**Comment by maru801** [comment]
> Slightly unrelated, but lets say we do sort singles into folders, I personally dont like there is no separation between Album and Single simply based on it being a single song, an option to allow this would be great EDIT: I have found the [feature request on Fider](https://features.jellyfin.org/posts/2220/group-music-releases-by-type-album-single-live-compilation-etc)

Album views in music player apps or servers "abuse" albums to have it be a catch-all place for all music collection types. Singles don't always come with a literal single audio track. Singles can come with multiple audio tracks. Singles are the same as albums. It's just that they have different thresholds for what is considered one or the other. 

As for an option to allow that behavior, maybe a toggle to make anything with a single audio track not have an "album" view could be made. It would also mean that if someone uploads an incomplete release to their server, incomplete releases with a single track would not get an album view when possibly the user wanted it to still get an album view (as they probably don't consider a single song from an album to also be a single). I guess you could also add more metadata info to the server to have users mark those tracks to be ignored by this toggle feature I'm made up, but that would no longer be automatic and rely on user input from then on. I believe this would work better as a client feature that just skips showing you albums with a single audio track instead of changing how the server stores media in its database.

> (I don't like it when a comment with a genuine question gets hidden as off topic).
> 
> I rephrase the question: Are there hurdles/problems/issues that keep us from implementing this request? Is there a particular reason why we have not implemented this request yet? The issue is 5 years old.

Your original post that got hidden for being off-topic doesn't state any question.

To answer the question, there just hasn't been any desire from the main devs to "hurry up" progress on this. That is not to say they don't care about this. They just have been focusing on other things. From my understanding, they want to redo most of the way music is handled before having new additions to it be worked on.

**Comment by hyperdotbat** [comment]
> As for an option to allow that behavior, maybe a toggle to make anything with a single audio track not have an "album" view could be made. It would also mean that if someone uploads an incomplete release to their server, incomplete releases with a single track would not get an album view when possibly the user wanted it to still get an album view (as they probably don't consider a single song from an album to also be a single). I guess you could also add more metadata info to the server to have users mark those tracks to be ignored by this toggle feature I'm made up, but that would no longer be automatic and rely on user input from then on. I believe this would work better as a client feature that just skips showing you albums with a single audio track instead of changing how the server stores media in its database.

With Vorbis tags there is a field called Total Tracks, which is self explanatory, and this could be used to decipher if its a song from an incomplete album.
And Im sorry to bring an unrelated issue, please point me if there is one already for this as I couldnt find one myself; why is Date not parsed fully, meaning `2025-06-10` becomes January 1st 2025 for the release date in clients such as Feishin

**Comment by davidl77** [comment]
I agrre with ; https://github.com/jellyfin/jellyfin/issues/6694#issuecomment-1344125305
I have same issue.

This is pretty common for people having a library with lot of "singles" insted od album!

IMHO **Jellyfin should rely on Metadata (album TAG) instead of folder!!!**
